### PR TITLE
Added ability to consume TemplateResponse for Step Functions API Gateway

### PR DESCRIPTION
**Pull request by ambasta**
Allows for MappginTemplate in Integration Reponse as:

```
stepFunctions:
  stateMachines:
    helloStep:
      events:
        - http:
            path: hello
            method: POST
            response:
              statusCodes:
                200:
                  pattern: ''
                  template: "#set($inputRoot = $input.path('$'))\n#set($arn = $inputRoot.executionArn)\n#set($lst = $arn.split(\":\"))\n#set($elem = $lst[-1])\n{\"job_id\": $util.escapeJavaScript($elem)}"
                  headers:
                    Content-Type: "'application/json'"
```

allowing to transform response to for example
`{"job_id": 3305683b-8106-11e9-b697-8bac36c37dad}`

### CODE REVIEW

**Comment by theburningmonk** [comment]
@ambasta can you take a look at what's failing on CI? looks like some linting errors

**Comment by theburningmonk** [comment]
@ambasta also, can you add some tests for this as well?

File: lib/deploy/events/apiGateway/methods.js [hunk 1]
```diff
@@ -178,13 +178,13 @@ module.exports = {
...
           ResponseParameters: {},
-          ResponseTemplates: {},
+          ResponseTemplates: [ this.getIntegrationResponseTemplates(200, http) ],
         },
         {
           StatusCode: 400,
           SelectionPattern: 400,
           ResponseParameters: {},
-          ResponseTemplates: {},
+          ResponseTemplates: [ this.getIntegrationResponseTemplates(200, http) ],
```
Comment (theburningmonk) [inline lib/deploy/events/apiGateway/methods.js]: shouldn't this be `400` instead of `200`?

File: lib/deploy/events/apiGateway/methods.js [hunk 2]
```diff
@@ -236,6 +236,22 @@ module.exports = {
     return defaultTemplate;
   },
 
+  getIntegrationResponseTemplates(StatusCode, http) {
+    const defaultTemplate = {};
+
+    if (_.has(http, "response.statusCodes")) {
```
Comment (theburningmonk) [inline lib/deploy/events/apiGateway/methods.js]: how about `if(_.has(http, `response.statusCodes.${StatusCode}.template`))`? or 
```
const template = _.get(http, `response.statusCodes.${StatusCode}.template`);
if (template) {
  return Object.assign(defaultTemplate, {'application/json': template});
} else {
  return defaultTemplate;
}
```

Comment (ambasta) [reply lib/deploy/events/apiGateway/methods.js]: Sure. I have never programmed in node, so I am trying to figure out how to improve on codebase (or write test cases for that matter)

**Comment by theburningmonk** [comment]
@ambasta did you get a chance to look at the build errors? looks like some linting errors:

image

and the PR is still missing tests

**Comment by ambasta** [comment]
> @ambasta did you get a chance to look at the build errors? looks like some linting errors:
> 
> image
> 
> and the PR is still missing tests

Yeah I did. I am trying to learn ts atm so I can write required test cases. I can either close this PR and push again at a later date or keep this open for changes to come through when they're ready?

**Comment by theburningmonk** [comment]
@ambasta OK, in that case, shall we close the PR for now? Just submit again when you're ready.

Not a fan of having long lived PRs ;-)

**Comment by ambasta** [comment]
Aight, closed for now.. Should re-open hopefully by EoW

**Pull request closed without merging**